diff/diff_test.go

```diff
@@ -50,6 +50,28 @@ func TestStripKey(t *testing.T) {
 	}
 }
 
+func TestDiffIntNullable(t *testing.T) {
+	var testcases = []struct {
+		desc         string
+		left         interface{}
+		right        interface{}
+		expectedDiff interface{}
+	}{
+		{"both nil", nil, nil, nil},
+		{"nil, one", nil, 1, 1},
+		{"one, nil", 1, nil, []interface{}{nil}},
+		{"both one", 1, 1, nil},
```

Comment:
should we also test diffing int pointer to int?

Reply:
With this change, don't we eliminate that as valid input to the diff layer?

Reply:
oh yes, good point. I had the wrong mindset reading this test.